docker/start.sh

```diff
@@ -134,9 +139,25 @@ wait_for_db() {
     fi
 }
 
+register_default_domain() {
+    echo "Temporal CLI Address: $TEMPORAL_CLI_ADDRESS"
+    sleep 5
+    echo "Registering default domain: $DEFAULT_DOMAIN_NAME"
+    until tctl --do $DEFAULT_DOMAIN_NAME domain describe < /dev/null; do
+        echo "Default domain $DEFAULT_DOMAIN_NAME not found.  Creating..."
+        sleep 1
+        tctl --do $DEFAULT_DOMAIN_NAME domain register --rd $DEFAULT_DOMAIN_RETENTION --desc "Default domain for Temporal Server"
+    done
+    echo "Default domain registration complete."
+}
+
 wait_for_db
 if [ "$SKIP_SCHEMA_SETUP" != true ]; then
     setup_schema
 fi
 
+if [ "$SKIP_DEFAULT_DOMAIN_CREATION" != true ]; then
+    register_default_domain &
```

Comment:
i am confused about how this works -- the server is not yet started, it won't actually start until line 163... or am i mis-reading this? @samarabbas